### DISCUSSION TITLE
make returned with non-zero (2) status

### DISCUSSION THREAD

**Discussion by hesselmans** [dtcenter/METplus#1944] [Installation] [answered]
./compile_MET_all.sh install_met_env.srv02 
terminates with the following message: make returned with non-zero (2) status
No executable is produced.

Logging can be found in error_MET.txt.
Furthermore error_MET.txt contains response to:
python3-config --cflags
find /usr -name "python3.10*"
python3-config --ldflags
The latter has been added, because I found a similar question posed previously and this info was asked for.

OS is:
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy

[error_MET.txt](https://github.com/dtcenter/METplus/files/10021639/error_MET.txt)
[make.log](https://github.com/dtcenter/METplus/files/10021650/make.log)
[install_met_env.srv02.txt](https://github.com/dtcenter/METplus/files/10021662/install_met_env.srv02.txt)

**Comment by jprestop** [comment]
HI @hesselmans.

Thank you for looking for similar Discussions regarding this issue and thank you for attaching these files.  I would like to take a look at two other files.  

Could you please send /home/wrf/repos/MET/v1/external_libs/bufrlib/BUFRLIB_v11.3.0.make.log so that we can ensure that the BUFR library successfully compiled?

Could you also please send your config.log file from the MET compilation attempt?

Unfortunately, I am leaving the office until 11/18 starting this afternoon.  If you are able to send those soon, that would be great.  If not, @JohnHalleyGotway will hopefully be able to provide assistance.

**Comment by hesselmans** [comment, marked as answer]
[config.log](https://github.com/dtcenter/METplus/files/10029093/config.log)
[make.log](https://github.com/dtcenter/METplus/files/10029096/make.log)

**Reply by jprestop** [reply]
Thank you for attaching these files, @hesselmans, and my apologies for the unusually long delay in responding to you.

In make.log, I do see that the BUFR library did not compile successfullly:
```
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (rank-1 and scalar)
```

In your install_met_env.srv02.txt file, please change:
```
export COMPILER=gnu
```
to 
```
export COMPILER=gnu_11.3.0
```
That should trigger this [line](https://github.com/dtcenter/MET/blob/61e192883f76bf7362152c19c2d279fa9d5dc5c7/scripts/installation/compile_MET_all.sh#L377) in the compile_MET_all.sh script to compile the BUFR lib in a way that is needed for gcc versions greater than 10.

Then, please rerun the script with your modified install_met_env.srv02.txt file.  I believe that should do the trick.  Please let us know how it goes.

**Reply by jprestop** [reply]
HI @hesselmans.  I just wanted to follow up on this discussion to see if questions remain. If not, please feel free to select the "Mark As Answer" button, and then I'll lock it to prevent future posts. If I don't hear otherwise in the next week or so, I'll just go ahead and mark it as being answered.  Thanks!
